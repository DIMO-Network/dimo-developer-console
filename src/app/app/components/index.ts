export * from './OnboardingCard';
export * from './Onboarding';
