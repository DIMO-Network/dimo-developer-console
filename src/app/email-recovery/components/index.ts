export * from './EmailRecoveryForm';
export * from './CheckEmail';