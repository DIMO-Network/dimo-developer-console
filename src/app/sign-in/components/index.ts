export * from './OtpInputForm';
export * from './PasskeyLogin';
export * from './SignInMethodForm';
