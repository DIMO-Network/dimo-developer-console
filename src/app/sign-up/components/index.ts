export * from './BuildForForm';
export * from './CompanyInfoForm';
export * from './SignUpWith';
export * from './UserInfoForm';
