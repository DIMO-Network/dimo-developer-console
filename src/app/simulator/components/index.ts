export { View } from './View';
export { Header } from './Header';
export { SimulatedVehiclesSection } from './SimulatedVehiclesSection';
export { CreateSimulatedVehicleModal } from './CreateSimulatedVehicleModal';
