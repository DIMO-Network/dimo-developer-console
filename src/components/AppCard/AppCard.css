.app-card {
  @apply flex flex-row justify-between cursor-pointer items-center;

  .content {
    @apply flex flex-col gap-1;

    .title {
      @apply text-base;
    }

    .description {
      @apply text-xs text-white/50 capitalize;
    }
  }
}
