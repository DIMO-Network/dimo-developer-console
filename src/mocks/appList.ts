import { IApp } from '@/types/app';

export const appListMock: IApp[] = [
  {
    name: 'Test App',
    scope: 'sandbox',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
  {
    name: 'Fleet Management App',
    scope: 'production',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
  {
    name: 'Data Analytics',
    scope: 'production',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
  {
    name: 'Hello World',
    scope: 'production',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
  {
    name: 'Hello World',
    scope: 'production',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
  {
    name: 'Hello World',
    scope: 'production',
    client_id: '0x*HSD)BN*(SHJDM)S(IJD SIOPDJSN)abcded1EA',
    signers: [
      {
        wallet: '0x1ABC7154748D1CE5144478CDEB574AE244B939B5',
        key: 'abcdef12345abcdef12345abcdef12345abcdef12345',
      },
    ],
    redirectUris: [
      {
        redirectUri: 'https://google.com/',
        status: 'active',
      },
      {
        redirectUri: 'https://google.com.co/',
        status: 'inactive',
      },
    ],
  },
];
