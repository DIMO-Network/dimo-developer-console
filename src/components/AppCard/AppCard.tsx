import { type FC } from 'react';
import classNames from 'classnames';

import { BeachAccessIcon, DeveloperBoardIcon } from '@/components/Icons';
import { Card } from '@/components/Card';
import { ENVIRONMENTS_LABELS, IApp } from '@/types/app';

import './AppCard.css';

interface IProps extends Partial<IApp> {
  className?: string;
  description?: string;
  onClick?: () => void;
}

const AppIcon = {
  production: <DeveloperBoardIcon className="w-5 h-5" />,
  sandbox: <BeachAccessIcon className="w-5 h-5" />,
};

export const AppCard: FC<IProps> = ({
  name,
  scope = 'production',
  description = '',
  className = '',
}) => {
  return (
    <Card className={classNames('app-card card-border', className)}>
      <div className="content">
        <p className="title">{name}</p>
        <p className="description">
          {description || ENVIRONMENTS_LABELS[scope]}
        </p>
      </div>
      {AppIcon[scope || 'sandbox']}
    </Card>
  );
};

export default AppCard;
