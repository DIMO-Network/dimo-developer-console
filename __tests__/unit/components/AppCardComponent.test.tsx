import { AppCard } from '@/components/AppCard';
import { appListMock } from '@/mocks/appList';
import { render, screen } from '@testing-library/react';

describe('AppCard', () => {
  it('renders an app card', () => {
    const [app] = appListMock;
    const { container } = render(<AppCard {...app} />);

    const appNameElm = screen.getByText('Test App');
    const environmentElm = screen.getByText('sandbox');

    expect(appNameElm).toBeInTheDocument();
    expect(environmentElm).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
