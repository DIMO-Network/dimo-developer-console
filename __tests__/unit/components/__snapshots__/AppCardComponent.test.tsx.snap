// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppCard renders an app card 1`] = `
<div>
  <div
    class="card app-card card-border"
  >
    <div
      class="content"
    >
      <p
        class="title"
      >
        Test App
      </p>
      <p
        class="description"
      >
        Sandbox
      </p>
    </div>
    <svg
      class="w-5 h-5"
      fill="none"
      role="beach-access-icon"
      viewBox="0 0 18 18"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.3403 11.7689L11.7703 10.3389L17.5003 16.0689C17.8903 16.4589 17.8903 17.0989 17.5003 17.4989C17.1103 17.8889 16.4703 17.8889 16.0703 17.4989L10.3403 11.7689ZM14.6303 6.03886L15.9003 4.76886C16.7903 3.87886 16.6703 2.33886 15.5903 1.68886C11.7003 -0.691138 6.56026 -0.201138 3.19026 3.15886C7.12026 1.85886 11.5003 2.90886 14.6303 6.03886ZM3.16026 3.18886C-0.199739 6.55886 -0.689739 11.6989 1.68026 15.5889C2.34026 16.6689 3.87026 16.7989 4.76026 15.8989L6.03026 14.6289C2.91026 11.4989 1.86026 7.11886 3.16026 3.18886ZM3.18026 3.16886L3.17026 3.17886C2.79026 6.18886 4.34026 10.0589 7.47026 13.1989L13.2003 7.46886C10.0703 4.33886 6.19026 2.78886 3.18026 3.16886Z"
        fill="white"
      />
      <path
        d="M10.3403 11.7689L11.7703 10.3389L17.5003 16.0689C17.8903 16.4589 17.8903 17.0989 17.5003 17.4989C17.1103 17.8889 16.4703 17.8889 16.0703 17.4989L10.3403 11.7689ZM14.6303 6.03886L15.9003 4.76886C16.7903 3.87886 16.6703 2.33886 15.5903 1.68886C11.7003 -0.691138 6.56026 -0.201138 3.19026 3.15886C7.12026 1.85886 11.5003 2.90886 14.6303 6.03886ZM3.16026 3.18886C-0.199739 6.55886 -0.689739 11.6989 1.68026 15.5889C2.34026 16.6689 3.87026 16.7989 4.76026 15.8989L6.03026 14.6289C2.91026 11.4989 1.86026 7.11886 3.16026 3.18886ZM3.18026 3.16886L3.17026 3.17886C2.79026 6.18886 4.34026 10.0589 7.47026 13.1989L13.2003 7.46886C10.0703 4.33886 6.19026 2.78886 3.18026 3.16886Z"
        fill="url(#paint0_radial_1956_55)"
        fill-opacity="0.6"
      />
      <defs>
        <radialgradient
          cx="0"
          cy="0"
          gradientTransform="translate(12.4341 7.79946) rotate(174.392) scale(12.2849 56.7031)"
          gradientUnits="userSpaceOnUse"
          id="paint0_radial_1956_749"
          r="1"
        >
          <stop
            offset="0.260417"
            stop-color="#6EE4DA"
            stop-opacity="0.8"
          />
          <stop
            offset="1"
            stop-color="#B1E3FF"
          />
        </radialgradient>
      </defs>
    </svg>
  </div>
</div>
`;
